app/controllers/orders_controller.rb

```diff
@@ -336,36 +302,61 @@ def index
 
   private
 
-  def update_order_details
-    # don't run if no updates for order_details
-
-    order_detail_updates = {}
-    params.each do |key, value|
-      if /\A(quantity)(\d+)\z/ =~ key && value.present?
-        order_detail_updates[Regexp.last_match(2).to_i] ||= {}
-        order_detail_updates[Regexp.last_match(2).to_i][Regexp.last_match(1).to_sym] = value
-      end
-      if /\A(note)(\d+)\z/ =~ key
-        order_detail_updates[Regexp.last_match(2).to_i] ||= {}
-        order_detail_updates[Regexp.last_match(2).to_i][Regexp.last_match(1).to_sym] = value
-      end
-    end
-
-    @order.update_details(order_detail_updates)
-  end
-
   def build_order_date
     if params[:order_date].present? && params[:order_time].present?
       parse_usa_date(params[:order_date], join_time_select_values(params[:order_time]))
     end
   end
 
+  def can_switch_instrument_on?
+    first_order_detail.reservation.can_switch_instrument_on?
+  end
+
+  def facility_ability
+    @facility_ability ||= Ability.new(session_user, @order.facility, self)
+  end
+
+  def first_order_detail
+    @first_order_detail ||= @order.order_details.first
+  end
+
   def load_statuses
     @order_statuses = OrderStatus.non_protected_statuses(@order.facility)
   end
 
-  def should_send_notification?
-    !acting_as? || params[:send_notification] == "1"
+  def order_purchaser
+    @order_purchaser ||= OrderPurchaser.new(
+      acting_as: acting_as?,
+      order: @order,
+      order_in_past: facility_ability.can?(:order_in_past, @order),
+      params: params,
+      user: session_user,
+    )
+  end
+
+  def order_update_params
+    @order_update_params ||= OrderDetailUpdateParamHashExtractor.new(params).to_h
+  end
+
+  def ordering_on_behalf_with_date_params?
+    params[:order_date].present? && params[:order_time].present? && acting_as?
+  end
+
+  def single_reservation?
+    @order.order_details.size == 1 &&
+      @order.order_details.first.product.is_a?(Instrument) &&
+      !@order.order_details.first.bundled? &&
+      !acting_as?
```

Comment:
It feels like the `acting_as?` check doesn't belong in this method. Would it make sense to pull the rest of the method into `Order`?